### PR TITLE
bump trivy to v0.57.1

### PR DESCRIPTION
**Pull request by nikpivkin**
In this PR:
- trivy updated to v0.57.1
- added a package with flags and options since they were removed from Trivy https://github.com/aquasecurity/trivy/pull/6995
- the plugin only uses Rego checks
- added support for `--no-progress` flag to suppress progressbar
- the log package from trivy (slog) is used for logging.

### CODE REVIEW

File: pkg/commands/run_test.go [hunk 1]
```diff
@@ -118,13 +182,41 @@ const expectedS3ScanResult = `{
             }
           }
         },
+        {
```
Comment (nikpivkin) [inline pkg/commands/run_test.go:185]: Strange that this check didn't work before, but according to the [s3onlycache.json](https://github.com/aquasecurity/trivy-aws/blob/main/pkg/commands/testdata/s3onlycache.json#L192-L229)  cache file the bucket has no logging.

File: pkg/commands/run_test.go [hunk 2]
```diff
@@ -166,36 +260,15 @@ const expectedS3ScanResult = `{
             }
           }
         },
-        {
```
Comment (nikpivkin) [inline pkg/commands/run_test.go:169]: This check is still considered [passed](https://github.com/aquasecurity/trivy-aws/pull/263/files#diff-ef103af943920a01839a075b421bdcff3319c58012056e739e5b3d048ae8389fR50-R71), but due to the specificity of Rego checks, all passed checks will have an empty target.

File: pkg/commands/run_test.go [hunk 3]
```diff
@@ -166,36 +260,15 @@ const expectedS3ScanResult = `{
...
-          "PrimaryURL": "https://avd.aquasec.com/misconfig/avd-aws-0092",
-          "References": [
-            "https://avd.aquasec.com/misconfig/avd-aws-0092"
-          ],
-          "Status": "PASS",
-          "Layer": {},
-          "CauseMetadata": {
-            "Resource": "arn:aws:s3:::examplebucket",
-            "Provider": "aws",
-            "Service": "s3",
```
Comment (nikpivkin) [inline pkg/commands/run_test.go:186]: This check is still considered [passed](https://github.com/aquasecurity/trivy-aws/pull/263/files#diff-ef103af943920a01839a075b421bdcff3319c58012056e739e5b3d048ae8389fR50-R71), but due to the specificity of Rego checks, all passed checks will have an empty target.

**Comment by nikpivkin** [comment]
@simar7 What do you think about if we move the JSON content of reports from go to separate golden files to make them easier to maintain?

**Comment by simar7** [comment]
> @simar7 What do you think about if we move the JSON content of reports from go to separate golden files to make them easier to maintain? I think we can also make the tests in [run_test.go](https://github.com/aquasecurity/trivy-aws/blob/main/pkg/commands/run_test.go) integration and set options via flags, as is done in [Trivy](https://github.com/aquasecurity/trivy/blob/main/integration/repo_test.go#L451-L475). I can do this in a separate PR.

Sure, sounds good to me.

File: pkg/report/report.go [hunk 4]
```diff
@@ -83,6 +83,9 @@ func Write(ctx context.Context, rep *Report, opt flag.Options, fromCache bool) e
 		return err
 	}
 
+	// combine results without a target for consistency of the result
+	filtered = combineResults(filtered)
```
Comment (simar7) [inline pkg/report/report.go:87]: What's this for? Should we do it as a separate feature PR?

Comment (nikpivkin) [reply pkg/report/report.go:87]: Results can have multiple records without a target in different order which sometimes causes tests to fail. This function simply merges the results without a target and sorts the misconfigs by AVDID.

**Review by simar7 — APPROVED** [review]

**Pull request merged**